connectivity/factory/factory.go

```diff
@@ -0,0 +1,248 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package factory
+
+import (
+	_ "embed"
+
+	"github.com/blang/semver/v4"
+
+	"github.com/cilium/cilium-cli/connectivity/check"
+	"github.com/cilium/cilium-cli/connectivity/factory/manifests/template"
+	"github.com/cilium/cilium-cli/utils/features"
+)
+
+var (
+	//go:embed manifests/allow-all-egress.yaml
+	allowAllEgressPolicyYAML string
+
+	//go:embed manifests/allow-all-ingress.yaml
+	allowAllIngressPolicyYAML string
+
+	//go:embed manifests/deny-all-ingress.yaml
+	denyAllIngressPolicyYAML string
+
+	//go:embed manifests/deny-cidr.yaml
+	denyCIDRPolicyYAML string
+
+	//go:embed manifests/allow-cluster-entity.yaml
+	allowClusterEntityPolicyYAML string
+
+	//go:embed manifests/client-egress-only-dns.yaml
+	clientEgressOnlyDNSPolicyYAML string
+
+	//go:embed manifests/client-egress-to-fqdns-one-one-one-one.yaml
+	clientEgressToFQDNsCiliumIOPolicyYAML string
+
+	//go:embed manifests/echo-ingress-from-other-client.yaml
+	echoIngressFromOtherClientPolicyYAML string
+
+	//go:embed manifests/client-egress-to-cidr-cp-host-knp.yaml
+	clientEgressToCIDRCPHostPolicyYAML string
+
+	//go:embed manifests/client-egress-to-cidr-external.yaml
+	clientEgressToCIDRExternalPolicyYAML string
+
+	//go:embed manifests/client-egress-to-cidr-external-knp.yaml
+	clientEgressToCIDRExternalPolicyKNPYAML string
+
+	//go:embed manifests/client-egress-to-cidr-k8s.yaml
+	clientEgressToCIDRK8sPolicyYAML string
+
+	//go:embed manifests/client-egress-to-cidr-node-knp.yaml
+	clientEgressToCIDRNodeKNPYAML string
+
+	//go:embed manifests/client-egress-to-cidr-external-deny.yaml
+	clientEgressToCIDRExternalDenyPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-http.yaml
+	clientEgressL7HTTPPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-http-named-port.yaml
+	clientEgressL7HTTPNamedPortPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-tls.yaml
+	clientEgressL7TLSPolicyYAML string
+
+	//go:embed manifests/client-egress-l7-http-matchheader-secret.yaml
+	clientEgressL7HTTPMatchheaderSecretYAML string
+
+	//go:embed manifests/echo-ingress-from-cidr.yaml
+	echoIngressFromCIDRYAML string
+)
+
+var (
+	clientLabel  = map[string]string{"name": "client"}
+	client2Label = map[string]string{"name": "client2"}
+
+	// Test order is important
```

Comment:
My bad.
I put this comment for two reasons:
1. To highlight that we want to have a predictable test run order (which is already guaranteed by slice).
2. Because of comments like this: https://github.com/cilium/cilium-cli/blob/main/connectivity/suite.go#L1261

Perhaps it's better to remove this comment?

Also, as the next step(s) I was thinking about splitting all the tests into the following chunks:
1. `before`: tests that must be run before other tests (for some reason) and run only sequentially in a single namespace.
2. `parallel`: tests that can be run concurrently in separate namespaces.
4. `after`: tests that must be run before other tests (for some reason) and run only sequentially in a single namespace.

Reply:
yeah something along that line makes sense

Reply:
could you open a github issue to follow up on this so that we don't forget about it 🚀 🙏

Reply:
I have a roadmap issue in the private repo. Should I replicate it to the `cilium/cilium-cli` repo?